pyriemann/classification.py

```diff
@@ -903,4 +902,131 @@ def predict_proba(self, X):
         prob : ndarray, shape (n_matrices, n_classes)
             Probabilities for each class.
         """
-        return softmax(-self._predict_distances(X)**2)
+        return softmax(-self._predict_distances(X) ** 2)
+
+
+def class_distinctiveness(X, y, exponent=1, metric='riemann',
+                          return_num_denom=False):
+    r"""Measure class distinctiveness between classes of SPD matrices.
+
+    For two class problem, the class distinctiveness between class A
+    and B on the manifold of SPD matrices is quantified as:
+
+    .. math::
+        \mathrm{classDis}(A, B, p) = \frac{d \left(\bar{X}^{A},
+        \bar{X}^{B}\right)^p}
+        {\frac{1}{2} \left( \sigma_{X^{A}}^p + \sigma_{X^{B}}^p \right)}
+
+    where :math:`\bar{X}^{K}` is the center of class K, ie the mean of matrices
+    from class K (see :func:`pyriemann.utils.mean.mean_covariance`) and
+    :math:`\sigma_{X^{K}}` is the class dispersion, ie the mean of distances
+    between matrices from class K and their center of class
+    :math:`\bar{X}^{K}`:
+
+    .. math::
+        \sigma_{X^{K}} = \frac{1}{m} \sum_{i=1}^m d
+        \left(X_i, \bar{X}^{K}\right)^p
+
+    For more than two classes, it is quantified as:
+
+    .. math::
+        \mathrm{classDis}\left(\left\{K_{j}\right\}\right, p) =
+        \frac{\sum_{j=1}^{c} d\left(\bar{X}^{K_{j}}, \tilde{X}\right)^p}
+        {\sum_{j=1}^{c} \sigma_{X^{K_{j}}}^p}
+
+    where :math:`\tilde{X}` is the mean of centers of class of all :math:`c`
+    classes.
```

Comment:
Add that `p` in this formula is `exponent` parameter.

Reply:
Thank you so much for the comment! Indeed, this is essential and I should have noticed that.
I hope now everything is ready.